rotkehlchen/db/dbhandler.py

```diff
@@ -1217,9 +1218,9 @@ def remove_single_blockchain_accounts(
         # Needs to happen before the address is removed since removing the address
         # will also remove evmtx_address_mappings, thus making it impossible
         # to figure out which transactions are touched by this address
-        if blockchain == SupportedBlockchain.ETHEREUM:
+        if blockchain in {SupportedBlockchain.ETHEREUM, SupportedBlockchain.OPTIMISM}:
```

Comment:
We have started having this in places and it will come bite us in the ass in the future.

Can you add in types.py, much like we have `SUPPORTED_CHAIN_IDS` and `SUPPORTED_EVM_CHAINS` a `SUPPORTED_EVM_CHAINS_WITH_TRANSACTIONS`. So long as we have this avalanche exception we need to have it easy to change in one place since if we don't we will otherwise have to search and replace stuff when we fix it.

Also use tuples instead of sets. And then use it here